Cargo.toml

```diff
@@ -13,6 +13,7 @@ members = [
 # This is useful for profiling and, to some extent, debug.
 # Besides, debug info should not affect the performance.
 debug = true
+incremental = true
```

Comment:
That's [the default](https://doc.rust-lang.org/cargo/reference/config.html#configuration-format) and we might want to disable it from time to time with an env var setting, not sure how this would interact with the env var and not sure I want to know now 🙂

Reply:
Is it by default? Tried to remove that line and see a slight increase in build time (+30sec) and cachepot shows different picture:
```
#25 5.453    Compiling prost-build v0.10.4
#25 7.517    Compiling tonic-build v0.7.2
#25 8.976    Compiling etcd-client v0.9.2
#25 16.03    Compiling etcd_broker v0.1.0 (/home/nonroot/libs/etcd_broker)
#25 17.48    Compiling pageserver v0.1.0 (/home/nonroot/pageserver)
#25 17.48    Compiling safekeeper v0.1.0 (/home/nonroot/safekeeper)
#25 100.4    Compiling control_plane v0.1.0 (/home/nonroot/control_plane)
#25 127.1     Finished release [optimized + debuginfo] target(s) in 2m 06s
#25 127.2 Compile requests                      15
#25 127.2 Compile requests executed              7
#25 127.2 Cache hits                             0
#25 127.2 Cache misses                           7
#25 127.2 Cache misses (Rust)                    7
#25 127.2 Cache timeouts                         0
#25 127.2 Cache read errors                      0
#25 127.2 Forced recaches                        0
#25 127.2 Cache write errors                     0
#25 127.2 Compilation failures                   0
#25 127.2 Cache errors                           7
#25 127.2 Cache errors (Rust)                    7
#25 127.2 Non-cacheable compilations             0
#25 127.2 Non-cacheable calls                    8
#25 127.2 Non-compilation calls                  0
#25 127.2 Unsupported compiler calls             0
#25 127.2 Average cache write               22.705 s
#25 127.2 Average cache read miss           22.705 s
#25 127.2 Average cache read hit             0.000 s
#25 127.2 Failed distributed compilations        0
#25 127.2
#25 127.2 Non-cacheable reasons:
#25 127.2 crate-type                             6
#25 127.2 -                                      2
#25 127.2
#25 127.2 Cache location                  S3, bucket: neon-github-dev
```
compared to this line ON:
```
#23 5.770    Compiling prost-build v0.10.4
#23 7.811    Compiling tonic-build v0.7.2
#23 9.250    Compiling etcd-client v0.9.2
#23 15.98    Compiling etcd_broker v0.1.0 (/home/nonroot/libs/etcd_broker)
#23 16.41    Compiling pageserver v0.1.0 (/home/nonroot/pageserver)
#23 16.41    Compiling safekeeper v0.1.0 (/home/nonroot/safekeeper)
#23 25.70    Compiling control_plane v0.1.0 (/home/nonroot/control_plane)
#23 64.03     Finished release [optimized + debuginfo] target(s) in 1m 03s
#23 64.12 Compile requests                     15
#23 64.12 Compile requests executed             3
#23 64.12 Cache hits                            0
#23 64.12 Cache misses                          3
#23 64.12 Cache misses (Rust)                   3
#23 64.12 Cache timeouts                        0
#23 64.12 Cache read errors                     0
#23 64.12 Forced recaches                       0
#23 64.12 Cache write errors                    0
#23 64.12 Compilation failures                  0
#23 64.12 Cache errors                          3
#23 64.12 Cache errors (Rust)                   3
#23 64.12 Non-cacheable compilations            0
#23 64.12 Non-cacheable calls                  12
#23 64.12 Non-compilation calls                 0
#23 64.12 Unsupported compiler calls            0
#23 64.12 Average cache write               2.468 s
#23 64.12 Average cache read miss           2.468 s
#23 64.12 Average cache read hit            0.000 s
#23 64.12 Failed distributed compilations       0
#23 64.12
#23 64.12 Non-cacheable reasons:
#23 64.12 crate-type                            6
#23 64.12 incremental                           4
#23 64.12 -                                     2
#23 64.12
#23 64.12 Cache location                  S3, bucket: neon-github-dev
```

So, is the release/incremental folder even used when `incremental = true` is unset?

Reply:
It should be enabled by default but I've just realized that we're using rustc 1.58.1 to build things, that might have some other default settings due to some bugs with incremental compilation, hence the difference.

Another explanation could be cachepot itself, that could cache interim results differently based on cargo options.

I would propose using `CARGO_INCREMENTAL` instead of editing the toml file instead, since the situation with this feature is relative volatile.

Reply:
Ok, `--build-arg CARGO_INCREMENTAL=0` can turn off the incremental build.